src/uct/api/uct.h

```diff
@@ -667,6 +698,70 @@ struct uct_iface_params {
 };
 
 
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Parameters for creating a UCT endpoint by @ref uct_ep_create
+ */
+struct uct_ep_params {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_ep_params_field. Fields not specified by this mask will be
+     * ignored.
+     */
+    uint64_t                          field_mask;
+
+    /**
+     * Interface to create the endpoint on. This is a mandatory field.
+     */
+    uct_iface_h                       iface;
+
+    /**
+     * User data associated with the endpoint.
+     */
+    void                              *user_data;
+
+    /**
+     * The device address to connect to on the remote peer. Should be defined
+     * together with @ref uct_ep_params_t::iface_addr to create an endpoint
+     * connected to a remote interface.
+     */
+    const uct_device_addr_t           *dev_addr;
+
+    /**
+      * Create an endpoint which is connected to remote interface.
+      * The interface address to connect to on the remote peer.
+      * @note requires @ref UCT_IFACE_FLAG_CONNECT_TO_IFACE capability.
+      */
+    const uct_iface_addr_t            *iface_addr;
+
+    /**
+     * The sockaddr to connect to on the remote peer. If set, @ref uct_ep_create
+     * will create an endpoint for a connection to the remote peer, specified by
+     * its socket address.
+     * @note The interface in this routine requires the
+     * @ref UCT_IFACE_FLAG_CONNECT_TO_SOCKADDR capability.
+     */
+    const ucs_sock_addr_t             *sockaddr;
+
+    /**
+     * @ref uct_cb_flags to indicate @ref uct_ep_params_t::sockaddr_pack_cb
+     * behavior. If @ref uct_ep_params_t::sockaddr_pack_cb is not set, this
+     * field will be ignored.
+     */
+    uint32_t                          sockaddr_cb_flags;
+
+    /**
+     * Callback for filling the user's private data which will be delivered to
+     * server side by @ref uct_sockaddr_conn_request_callback_t . This field is
+     * valid only if @ref uct_ep_params_t::sockaddr is set.
```

Comment:
Suggested re-wording:
Callback that will be used for filling the user's private data to be delivered to the server by `@ref` uct_sockaddr_conn_request_callback_t. This field is only valid if `@ref` uct_ep_params_t::sockaddr is set.